.travis.yml

```diff
@@ -34,6 +34,7 @@ deploy:
   provider: script
   script:
     - sbt -J-XX:ReservedCodeCacheSize=256m +publish
+    - sbt 'set credentials += Credentials("whitesource", "whitesourcesoftware.com", "", System.getenv("WHITESOURCE_KEY"))' whitesourceUpdate
```

Comment:
Is there a reason not to add the credentials line to the build proper (making it a no-op when the key is not defined)?

Reply:
I think the 'typical' use is to put the credential in your `~/.sbt/1.0/creds.sbt`, so this is an exceptional case. Might indeed be nice to add something to the build that turns that environment variable into a `Credential` if available, though, indeed